src/Symfony/Component/Form/ResolvedFormType.php

```diff
@@ -205,8 +205,16 @@ public function getOptionsResolver()
 
             $this->innerType->setDefaultOptions($this->optionsResolver);
 
+            if (method_exists($this->innerType, 'configureOptions')) {
```

Comment:
if someone does not use the AbstractType, they are required to implement setDefaultOptions because of the interface.

IMO, we should stick to the interface for the external usage of the FormTypeInterface, and build the forward-compatibility layer only in AbstractType, by making ``setDefaultOptions`` call ``configureOptions``. This will not break applications

If someone uses the FormTypeInterface without AbstractType, they will have a hard BC break for 3.0 when we will replace the method in the interface, but I haven't seen any code recreating the implementation from scratch rather than extending the base class with empty implementations.